spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,653 @@
+    :param slices_avg: Slices in PAM50 space to average metrics.
+    :return: ma: float64: Metric above the compression
+    :retrun: mb: float64: Metric below the compression
+    :retrun: mi: float64: Metric at the compression level
+    """
+    # find index of slices to average
+    idx_compression = df['Slice (I->S)'].isin(slices_avg).tolist()
+    idx_above = df['Slice (I->S)'].isin(z_range_above).tolist()
+    idx_below = df['Slice (I->S)'].isin(z_range_below).tolist()
+    ma = df.loc[idx_above, metric].mean()
+    mb = df.loc[idx_below, metric].mean()
+    mi = df.loc[idx_compression, metric].mean()
+    return ma, mb, mi
+
+
+def metric_ratio(ma, mb, mi):
+    """
+    Compute MSCC (Maximum Spinal Cord Compression) using the chosen metric of compression and of levels
+    above and bellow.
+    :param float: ma: metric of level above compression level.
+    :param float: mb: metric of level above compression level.
+    :param float: mi: metric at compression level.
+    :return float: metric ratio in %
+    """
+    return (1 - float(mi) / ((ma + mb) / float(2))) * 100
+
+
+def metric_ratio_norm(metrics_patients, metrics_HC):
+    """
+    Compute normalized MSCC (Maximum Spinal Cord Compression) using the chosen metric at the compression and
+    levels above and bellow.
+    Each metric is divided by the corresponding value in healthy controls.
+    :param list: metrics_patients: list metric value of level above, below and at compression of patient.
+    :param list: metrics_HC: list metric value of level above, below and at compression of healthy
+    controls.
+    :return float: MSCC normalized in %
+    """
+    ma = metrics_patients[0]/metrics_HC[0]
+    mb = metrics_patients[1]/metrics_HC[1]
+    mi = metrics_patients[2]/metrics_HC[2]
+    return metric_ratio(ma, mb, mi)
+
+
+def save_csv(fname_out, level, metric, metric_ratio, metric_ratio_norm, filename):
+    """
+    Save .csv file of MSCC results.
+    :param fname_out:
+    :param level: int: Level of compression.
+    :param metric: str: metric to perform normalization
+    :param metric_ratio: float:
+    :param metric_ratio_norm:
+    :param filename: str: input filename
+    :return:
+    """
+    if not os.path.isfile(fname_out):
+        with open(fname_out, 'w') as csvfile:
+            header = ['filename', 'Compression Level', metric + ' ratio', 'Normalized ' + metric + ' ratio']
+            writer = csv.DictWriter(csvfile, fieldnames=header)
+            writer.writeheader()
+    with open(fname_out, 'a') as csvfile:
+        csv_writer = csv.writer(csvfile, delimiter=',')
+        line = [filename, level, metric_ratio, metric_ratio_norm]
+        csv_writer.writerow(line)
+
+
+def main(argv: Sequence[str]):
+    parser = get_parser()
+    arguments = parser.parse_args(argv)
+    verbose = arguments.v
+    set_loglevel(verbose=verbose)    # values [0, 1, 2] map to logging levels [WARNING, INFO, DEBUG]
+
+    # Step 0: Argument validation
```

Comment:
Just to make things a tiny bit clearer, I think we can rename this :

```suggestion
    # Step 0: Argument loading and validation
```